hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -2467,27 +2468,66 @@ def test_not_equal(x, y):
         self.example_wise_shrink(Ordering, key=sort_key)
 
     def alphabet_minimize(self):
-        """Attempts to replace most bytes in the buffer with 0 or 1. The main
+        """Attempts to minimize the "alphabet" - the set of bytes that
+        are used in the representation of the current buffer. The main
         benefit of this is that it significantly increases our cache hit rate
         by making things that are equivalent more likely to have the same
-        representation.
-
-        We only run this once rather than as part of the main passes as
-        once it's done its magic it's unlikely to ever be useful again.
-        It's important that it runs first though, because it makes
-        everything that comes after it faster because of the cache hits.
+        representation, but it's also generally a rather effective "fuzzing"
+        step that gives us a lot of good opportunities to slip to a smaller
+        representation of the same bug.
         """
-        for c in (1, 0):
-            alphabet = set(self.buffer) - set(hrange(c + 1))
+        for c in range(255, 0, -1):
+            buf = self.buffer
 
-            if not alphabet:
+            if c not in buf:
                 continue
 
-            def clear_to(reduced):
-                reduced = set(reduced)
-                attempt = hbytes(
-                    [b if b <= c or b in reduced else c for b in self.buffer]
+            def try_replace(d):
```

Comment:
Minor: `try_replace(d)` is a bit confusing when `d` is the replacement value, not the value being replaced.

Something like `try_replace_with(d)` could be clearer.